ospf6d/ospf6_abr.c

```diff
@@ -53,6 +53,16 @@
 
 unsigned char conf_debug_ospf6_abr;
 
+int ospf6_ls_origin_same(struct ospf6_path *o_path, struct ospf6_route *route)
```

Comment:
nit: couldn't we change the second parameter to be a pointer to an `ospf6_path` struct as well? A comparison function that takes parameters of different types is a bit unconventional (ideally we'd have something like `path1` and `path2`).